### ISSUE TITLE
MusicGen on A100/A10G/3090 is Single Core CPU Bound

### ISSUE THREAD

**Issue by zaptrem** [facebookresearch/audiocraft#192]
Even with a batch size of one I'm getting results like this across the board. 
![image](https://github.com/facebookresearch/audiocraft/assets/1612230/ad0b6b50-fccb-46e3-90e8-521a0b63744c)

Is this something that can be fixed on my end? If not, what's the cause?

**Comment by iAlborz** [comment]
same issue on M1 Max. Only one core is being used.

**Comment by carlthome** [comment]
Interesting find! I too was confused by not seeing clear speedups when changing from a T4 to a V100 with the demo notebook. Just assumed the autoregressive nature of the model means there's a loop around the forward pass, unamenable to GPU parallelism.

**Comment by Pozaza** [comment]
You need to reinstall torch

**Comment by zaptrem** [comment]
@Pozaza @Redskull-127 All of that is managed by our cloud provider. However, seeing as our other models do not encounter this issue I think the cause is more likely related to the specifics of MusicGen/AudioCraft. Is there something special about MusicGen that relies on Intel's Python distro, for example?

**Comment by niatro** [comment]
Same Issue here for a 4090 GPU, is not been used. Only CPU.
![image](https://github.com/facebookresearch/audiocraft/assets/34010695/01f0ab03-f1eb-480e-86dd-80a4e8111b0e)

**Comment by zaptrem** [comment]
> Same Issue here for a 4090 GPU, is not been used. Only CPU. ![image](https://user-images.githubusercontent.com/34010695/260241338-01f0ab03-f1eb-480e-86dd-80a4e8111b0e.png)

This doesn't show much. You need to expose logical processors and use Afterburner or similar to track actual GPU usage.

**Comment by niatro** [comment]
Yep, u are right. Anyway I uninstalled audiocraft and installed again, I made sure that I create a good environment with conda. Unfortunately the guide in the repository is not straight forward but finally after I made an environment with Python 3.9, PyTorch 2.0.0 and ffmpeg and cloned the repository again and all the project worked fine. This issue is closed for me.  Thanks

**Comment by Redskull-127** [comment]
@zaptrem strange :(

**Comment by zaptrem** [comment]
> Yep, u are right. Anyway I uninstalled audiocraft and installed again, I made sure that I create a good environment with conda. Unfortunately the guide in the repository is not straight forward but finally after I made an environment with Python 3.9, PyTorch 2.0.0 and ffmpeg and cloned the repository again and all the project worked fine. This issue is closed for me.  Thanks


Can you post a screenshot of your logical processor and GPU utilization graphs during inference?

**Comment by mepc36** [comment]
I am seconding @zaptrem's request to @niatro to please post the following, it'd be a huge help!

> Can you post a screenshot of your logical processor (e.g., individual cores/hyperthreads) and GPU utilization graphs during inference?

**Comment by niatro** [comment]
Do you mean this graph?
![image](https://github.com/facebookresearch/audiocraft/assets/34010695/3dc2be54-dc0f-42ae-9897-282b0badca36)

And this graph?
![image](https://github.com/facebookresearch/audiocraft/assets/34010695/837901e8-cd25-41c5-a9e4-aa48a466a384)

During inference time

**Comment by Redskull-127** [comment]
> Yep, u are right. Anyway I uninstalled audiocraft and installed again, I made sure that I create a good environment with conda. Unfortunately the guide in the repository is not straight forward but finally after I made an environment with Python 3.9, PyTorch 2.0.0 and ffmpeg and cloned the repository again and all the project worked fine. This issue is closed for me. Thanks

looks like you're right man! Thanks for the support.

**Comment by zaptrem** [comment]
@niatro Close, can you right click the CPU graph and select Change graph to > Logical processors? I'm trying to figure out what your single core utilization looks like. Also, are you running this directly on Windows or using WSL?

**Comment by zeke-john** [comment]
Any update? Currently facing the same issue.
